jaas/src/main/java/com/griddynamics/jagger/jaas/rest/HttpGetResponseProducer.java

```diff
@@ -0,0 +1,53 @@
+package com.griddynamics.jagger.jaas.rest;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+
+/**
+ * Produces {@link org.springframework.http.ResponseEntity} response with two possible results:
+ * <ul>
+ * <li>{@link org.springframework.http.HttpStatus#OK} if there is payload to return </li>
+ * <li>{@link org.springframework.http.HttpStatus#NOT_FOUND} if there is nothing to return </li>
+ * </ul>
+ */
+public class HttpGetResponseProducer<T, R> {
+    
+    private final T responseSource;
+    private final Function<T, R> responseFunction;
+    
+    public HttpGetResponseProducer(T responseSource, Function<T, R> responseFunction) {
+        this.responseSource = responseSource;
+        this.responseFunction = responseFunction;
+    }
+    
+    public static <T, R> ResponseEntity<R> produce(T responseSource, Function<T, R> responseFunction) {
+        return new HttpGetResponseProducer<>(responseSource, responseFunction).produce();
+    }
+    
+    public ResponseEntity<R> produce() {
+        if (responseSource != null) {
+            R response = responseFunction.apply(responseSource);
+            if (Objects.nonNull(response)) {
+                if (isEmptyCollection(response)) {
```

Comment:
What if you get not a Collection, but a simple array? Should it be covered also?